players.js

```diff
@@ -259,9 +259,43 @@ class Player {
 			this.proxy.PreviousRemote()
 	}
 	activatePlayer(){
-		if(this.desktopApp){
-			let app = Shell.AppSystem.get_default().lookup_app(this.desktopApp);
-			app.activate();
+		let focusedWindow = global.display.get_focus_window();
+		let playerWindow = this._guessAppWindow(this.identity);
+
+		if (!playerWindow)
+			return
+
+		if (focusedWindow == playerWindow){
+			if (this.playerWindowMinimized)
+				playerWindow.minimize();
+
+			let apps = Shell.AppSystem.get_default().get_running();
+			apps[1].activate(global.get_current_time());
+		}
+		else{
+			if(this.desktopApp){
+				this.playerWindowMinimized = playerWindow.minimized;
+				Shell.AppSystem.get_default().lookup_app(this.desktopApp).activate();
+			}
+		}
+	}
+	_guessAppWindow(identity){//secondary best-guess method to match player with window
+		identity = identity.toLowerCase();
+		let windows_list = global.get_window_actors();
+		windows_list = windows_list.filter(element => element.get_meta_window().get_window_type() == 0); //only keep normal windows
+		
+		for (const actor of windows_list) {//first pass exact match
+			const window = actor.get_meta_window();
+			let wm_class = window.get_wm_class().toLowerCase();
+			if (wm_class == identity)
+				return window;
+		}
+		
+		for (const actor of windows_list) {//fallback approximate match
+			const window = actor.get_meta_window();
+			let wm_class = window.get_wm_class().toLowerCase();
+			if (wm_class.includes(identity) | identity.includes(wm_class))
```

Comment:
the `|` operator is the bitwise OR operator instead of the logical OR operator (`||`)  
I'm not sure if they cause the same effect

Reply:
Good point, I agree that it should be changed